lib/mocha/expectation.rb

```diff
@@ -557,32 +556,31 @@ def match?(actual_method_name, *actual_parameters)
 
     # @private
     def invocations_allowed?
-      @cardinality.invocations_allowed?(@invocation_count)
+      @cardinality.invocations_allowed?(@invocations.size)
     end
 
     # @private
     def satisfied?
-      @cardinality.satisfied?(@invocation_count)
+      @cardinality.satisfied?(@invocations.size)
     end
 
     # @private
     def invoke(*arguments)
-      @invocation_count += 1
       perform_side_effects
       invocation = Invocation.new(method_name, @yield_parameters, @return_values)
       @invocations << invocation
-      invocation.call(*arguments) { |*args| yield(*args) }
+      invocation.call(*arguments) { |*yield_args| yield(*yield_args) }
```

Comment:
Minor: The `args` -> `yield_args` rename appears to be orthogonal to the stated purpose of this commit. It would be good to extract it into its own commit.

Reply:
Resolved in https://github.com/freerange/mocha/pull/394#issuecomment-551927383.